common/include/pcl/common/io.h

```diff
@@ -337,21 +338,9 @@ namespace pcl
     * \note Assumes unique indices.
     * \ingroup common
     */
-  template <typename PointT> void 
-  copyPointCloud (const pcl::PointCloud<PointT> &cloud_in, 
-                  const std::vector<int> &indices, 
-                  pcl::PointCloud<PointT> &cloud_out);
- 
-  /** \brief Extract the indices of a given point cloud as a new point cloud
-    * \param[in] cloud_in the input point cloud dataset
-    * \param[in] indices the vector of indices representing the points to be copied from \a cloud_in
-    * \param[out] cloud_out the resultant output point cloud dataset
-    * \note Assumes unique indices.
-    * \ingroup common
-    */
-  template <typename PointT> void 
+  template <typename PointT, typename Allocator = std::allocator<int>> void
```

Comment:
Do I understand right that you are "merging" versions of this function with and without explicit allocator for indices?

I was wondering if `Allocator` is specific enough for template parameter name. Maybe `IndexAllocator`, so that nobody thinks that this has anything to do with points? Also, maybe add a suffix `T` in the end to match `PointT` style? I'm just testing the waters here and I am interested to hear your opinion.

Reply:
`IndexAllocator` sounds good. `IndexAllocatorT` implies of type T which it is not. As for being specific, I just copied it from the definition of `std::vector`

Reply:
> As for being specific, I just copied it from the definition of std::vector

Yeah, this is understood. But this function has more templated entities, so further clarification is needed IMHO.

> XAllocatorT implies type T which it is not (for the allocator).

Hm, I always assumed that `T` is just short for "type". That is, `PointT` means "point type". But anaway, I'm totally fine without `T`. This also matches STL style.

> IndicesVecAllocator sounds good to me.

Or `IndicesVectorAllocator`. But up to you.

Reply:
> PointT means "point type"

I always thought `PointT` was shorthand for type T but it's used for points (`PointRGBA`, `PointXYZ`).

Reply:
Well, perhaps you are right. But we'll never know for sure :smile: